File: web_client/types/lead.ts
/**
 * Lead Data Types
 * Golden Sync: mirrors backend_python/src/models/lead.py
 *
 * Backend models:
 *   - LeadData       → base customer data (no uid/session_id)
 *   - LeadDocument   → full Firestore document (extends LeadData, adds uid/session_id/metadata)
 *
 * MANDATORY: Any changes here MUST be reflected in backend_python/src/models/lead.py.
 */

/** Mirrors LeadData (backend_python/src/models/lead.py) */
export interface LeadData {
    name: string;
    email: string;
    phone?: string;
    project_details: string; // Mapped from quote_summary in the UI
    room_type?: string;      // Optional enrichment
    style?: string;          // Optional enrichment
}

/**
 * Mirrors LeadDocument (backend_python/src/models/lead.py)
 * Full Firestore document — includes uid, session_id and metadata.
 */
export interface LeadDocument extends LeadData {
    uid: string;
    session_id: string;      // Required for context tracking
    created_at: string;      // ISO 8601
    source: string;          // e.g. "chat"
}

export interface LeadSubmissionResponse {
    status: 'success' | 'error';
    message: string;
    request_id?: string;
}

File: backend_python/src/models/lead.py
from datetime import datetime

from pydantic import BaseModel, EmailStr, Field
from src.utils.datetime_utils import utc_now


class LeadData(BaseModel):
    """Customer lead data model."""
    model_config = {"extra": "forbid"}
    name: str = Field(..., max_length=100, description="Customer name")
    email: EmailStr = Field(..., description="Customer email address")
    phone: str | None = Field(None, max_length=20, description="Customer phone number")
    project_details: str = Field(..., max_length=2000, description="Detailed project description")
    room_type: str | None = Field(None, max_length=100, description="Type of room")
    style: str | None = Field(None, max_length=100, description="Preferred design style")

class LeadDocument(LeadData):
    """Complete lead document with metadata for Firestore."""
    uid: str = Field(..., description="User Firebase UID")
    session_id: str = Field(..., description="Chat session ID")
    created_at: datetime = Field(default_factory=utc_now, description="Timestamp")
    source: str = Field(default="chat", description="Lead source")
